src/terminal/Grid.h

```diff
@@ -577,30 +585,38 @@ void Grid<Cell>::render(RendererT&& _render, ScrollOffset _scrollOffset) const
     assert(!_scrollOffset || unbox<LineCount>(_scrollOffset) <= historyLineCount());
 
     auto y = LineOffset(0);
+    _isBlinkingVisible = false;
```

Comment:
I think the render() function really should remain const as it is and not generate any side effects that are later (e.g. in the future when somebody is reading the code) really hard to find and work with, maintainability wise.

What I'd suggest instead is constructing some kind of struct that will contain information that has been gathered during rendering, such as if the rendered grid cells do contain slow/rapid blinking cells and return that struct in Grid.render rather than nothing (void).

It is totally fine if such a struct initially only contains that one bool.
The struct helps readability/maintainability and future extensibility. I think a good use could be later on to indicate other meta information such as if any of the rendered lines do contain more complex unicode (to decide what optimized rendering path to chose from in the future) or any other meta information that would be expensive to be queried.

```cpp
struct RenderPassHints
{
    bool containsBlinkingCells = false;
};
```
Then, this struct could be returned by render() and its yielded information be used later on.